test/xpu_api/utilities/utility/utility.swap/swap_utility.pass.cpp

```diff
@@ -0,0 +1,165 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/utility>
+#include <oneapi/dpl/complex>
+
+#include "support/utils.h"
+#include "support/move_only.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct CopyOnly
+{
+    CopyOnly() = default;
+    CopyOnly(CopyOnly const&) noexcept = default;
+    CopyOnly&
+    operator=(CopyOnly const&)
+    {
+        return *this;
+    }
+};
+
+struct NoexceptMoveOnly
+{
+    NoexceptMoveOnly() = default;
+    NoexceptMoveOnly(NoexceptMoveOnly&&) noexcept = default;
+    NoexceptMoveOnly&
+    operator=(NoexceptMoveOnly&&) noexcept
+    {
+        return *this;
+    }
+};
+
+struct NotMoveConstructible
+{
+    NotMoveConstructible&
+    operator=(NotMoveConstructible&&)
+    {
+        return *this;
+    }
+
+    NotMoveConstructible(NotMoveConstructible&&) = delete;
+};
+
+struct NotMoveAssignable
+{
+    NotMoveAssignable(NotMoveAssignable&&) = delete;
+
+    NotMoveAssignable&
+    operator=(NotMoveAssignable&&) = delete;
+};
+
+template <class Tp>
+auto
+can_swap_test(int) -> decltype(dpl::swap(dpl::declval<Tp>(), dpl::declval<Tp>()));
+
+template <class Tp>
+auto
+can_swap_test(...) -> dpl::false_type;
+
+template <class Tp>
+constexpr bool
+can_swap()
+{
+    return dpl::is_same_v<decltype(can_swap_test<Tp>(0)), void>;
+}

```

Comment:
The same as for previous file, lets use `std::is_swappable`

Reply:
Done